arviz/data/inference_data.py

```diff
@@ -133,6 +133,18 @@ def __repr__(self):
             msg += "\n\nWarmup iterations saved ({}*).".format(WARMUP_TAG)
         return msg
 
+    def _repr_html_(self):
+        """Make html representation of object"""
+        xr.set_options(display_style="html")
+        html_template, element_template, css_template = html_templates(self)
+        template, elements = "", ""
+        for data_group in self._groups:
+            func = getattr(self, data_group)
+            elements += element_template.format(data_group, func._repr_html_())
+        template += html_template.format(elements)
```

Comment:
Are these `+=` operators adding strings together? If so could this be changed to another method of string concat? The reason I ask is that `+=` is typically used for numbers or other things and it's confusing to see it used as strings.

Its typically not used for string concats for the reasons detailed in the article below

https://wiki.python.org/moin/PythonSpeed/PerformanceTips#String_Concatenation

Reply:
I would also rename `func` to `ds`, `group` or `data` so the name is related to the object it stores. Maybe something like:

```
for group in self._groups_all:
    ds = getattr(...
```

I'd also use `_groups_all` given that groups start hidden and users can expand the desired ones

Reply:
@canyon289 Thanks for the suggestion. I'd keep it in mind for the future. I also learnt a lot of other tricks too from the article :)